thorlcr/activities/enth/thenthslave.cpp

```diff
@@ -94,20 +99,32 @@ class BaseEnthActivity : public CSlaveActivity, implements ILookAheadStopNotify
     {
         appendOutputLinked(this);
     }
-    virtual void setInputStream(unsigned index, CThorInput &_input, bool consumerOrdered) override
-    {
-        PARENT::setInputStream(index, _input, consumerOrdered);
-        originalInputStream = inputStream;
-    }
     virtual void start() override
     {
         ActivityTimer s(totalCycles, timeActivities);
         PARENT::start();
+        stopped = false;
         IHThorEnthArg *helper = static_cast <IHThorEnthArg *> (queryHelper());
         counter = 0;
+        lookAheadStarted = false;
+        enthStream = inputStream;
         denominator = validRC(helper->getProportionDenominator());
         numerator = validRC(helper->getProportionNumerator());
     }
+    virtual void stop() override
```

Comment:
@jakesmith 
I can see why you are doing it, but this code looks less clean than the previous code.  It seems to be heading in the wrong direction with the stranding code.

Could you possibly set inputStream to originalInputStream in start(), and have similar code to the new code to cache lookupHead, and assign it to inputStream?